mindsdb/api/executor/planner/query_planner.py

```diff
@@ -818,7 +821,64 @@ def from_query(self, query=None):
         else:
             raise PlanningException(f'Unsupported query type {type(query)}')
 
-        return self.plan
+        plan = self.handle_partitioning(self.plan)
+
+        return plan
+
+    def handle_partitioning(self, plan):
+        # handle fetchdataframe partitioning
+        steps_out = []
+
+        partition_step = None
+        for step in plan.steps:
+            if isinstance(step, FetchDataframeStep) and step.params is not None:
+                batch_size = step.params.get('batch_size')
+                if batch_size is not None:
+                    # found batched fetch
+                    partition_step = FetchDataframeStepPartition(
+                        step_num=step.step_num,
+                        integration=step.integration,
+                        query=step.query,
+                        raw_query=step.raw_query,
+                        params=step.params
+                    )
+                    steps_out.append(partition_step)
+                    # mark plan
+                    plan.is_resumable = True
+                    continue
+
+            if partition_step is not None:
+                # check and add step into partition
+
+                can_be_partitioned = False
+                if isinstance(step, (JoinStep, ApplyPredictorStep, InsertToTable)):
```

Comment:
Better to make `can_be_partitioned` as property of a step

Reply:
for partitioning fetching from database FetchDataframeStepPartition step is used
also the whole plan have to be marked to handle it in sql_query, it is done by:
`plan.is_resumable = True`